src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs

```diff
@@ -4,92 +4,236 @@
-                    {
-                        ProtocolVersion = protocol
-                    };
-
-                    var errorSink = new ErrorSink();
-                    var resolvedDescriptors = new List<TagHelperDescriptor>();
-                    for (var i = 0; i < assemblyNames.Values.Count; i++)
-                    {
-                        var assemblyName = assemblyNames.Values[i];
-                        var descriptors = descriptorResolver.Resolve(assemblyName, errorSink);
-                        resolvedDescriptors.AddRange(descriptors);
-                    }
-
-                    var resolvedResult = new ResolvedTagHelperDescriptorsResult
-                    {
-                        Descriptors = resolvedDescriptors,
-                        Errors = errorSink.Errors
-                    };
-                    var serializedResult = JsonConvert.SerializeObject(resolvedResult, Formatting.Indented);
-
-                    Console.WriteLine(serializedResult);
-
-                    return 0;
-                });
+                    config.OnExecute(
+                        () => Dispatch(
+                            projectArgument,
+                            assemblyNamesArgument,
+                            protocolOption,
+                            frameworkOption,
+                            configurationOption,
+                            buildBasePathOption));
+                }
+                else
+                {
+                    config.OnExecute(() => Run(assemblyNamesArgument, protocolOption));
+                }
             });
         }
 
-        public static ProjectContext ResolveProjectContext(string projectFilePath)
+        private static int Dispatch(
+            CommandArgument projectArgument,
+            CommandArgument assemblyNameArgument,
+            CommandOption protocolOption,
+            CommandOption frameworkOption,
+            CommandOption configurationOption,
+            CommandOption buildBasePathOption)
         {
-            var projectContexts = ProjectContext.CreateContextForEachFramework(projectFilePath);
-            var projectContext = projectContexts.FirstOrDefault();
+            var projectFilePath = projectArgument.Value;
+            var projectFile = ProjectReader.GetProject(projectFilePath);
+            var targetFrameworks = projectFile
+                .GetTargetFrameworks()
+                .Select(frameworkInformation => frameworkInformation.FrameworkName);
 
-            if (projectContext == null)
+            NuGetFramework framework;
+            if (!TryResolveFramework(frameworkOption, targetFrameworks, projectFilePath, out framework))
             {
-                throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentCulture, Resources.InvalidProjectFile, projectFilePath));
+                // Could not resolve framework for dispatch. Error was reported, exit early.
```

Comment:
Actually, this is logged in a different format within ResolveFramework, going to leave as is.